### PR TITLE
Feature/vote 1109/korean state character logic

### PR DESCRIPTION
**Pull request by SamiraMSadat**
## Jira ticket (required)
https://bixal-projects.atlassian.net/browse/VOTE-1109

## Description (optional)
Rework the Korean state character logic to use translation files

### Pre-deploy steps
1. lando retune

### Testing steps
1. Visit http://vote-gov.lndo.site/ko/register/as and verify the state char is not displayed (Korean - territory)
2. Visit http://vote-gov.lndo.site/ko/register/co and verify the state char is displayed (Korean - state)
3. Visit http://vote-gov.lndo.site/register/co and verify that the "state" string does not display (English - state)

### CODE REVIEW

File: web/themes/custom/vote_gov/templates/content/page-title.html.twig [hunk 1]
```diff
@@ -14,16 +14,7 @@
...
-{# set alt title for state pages #}
-{% if node_type == 'state_territory' %}
-  {% if language == 'ko' and currentnode.field_is_state.0.value == 1 %}
-    {# set title to alt text for state pages in korean #}
-    {% set title = 'Register to vote in @state_name state' | t ({"@state_name": title}) %}
-  {% else %}
-    {% set title = 'Register to vote in @state_name' | t ({"@state_name": title}) %}
-  {% endif %}
-{% endif %}
+{% set title = 'Register to vote in @state_name' | t ({"@state_name": title}) %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/content/page-title.html.twig:17]: Update this to add in the korean state character
```
{% if node_type == "state_territory" and language == 'ko' and currentnode.field_is_state.value %}
  {% set korean_state_char = "state" | t %}
  {% set title = title ~ " " ~ korean_state_char %}
{% endif %}

{% set title = 'Register to vote in @state_name' | t ({"@state_name": title}) %}
```

File: web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig [hunk 2]
```diff
@@ -9,6 +9,13 @@
 
 {% block content %}
 {% set reg_type = content.field_registration_type | field_value | column('#markup') %}
+{# Korean 'state' character logic #}
+{% if language == 'ko' and node.field_is_state.value %}
+  {% set korean_state_char = "state" | t %}
+  {% set state_name = (label | field_value | render)  ~ " " ~ korean_state_char %}
+{% else %}
+  {% set state_name = label | field_value | render %}
+{% endif %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/node/node--state-territory--full.html.twig:18]: Simplify this to remove redundancy
```
{% set state_name = label | field_value | render %}

{# Korean 'state' character logic #}
{% if language == 'ko' and node.field_is_state.value %}
  {% set korean_state_char = "state" | t %}
  {% set state_name = state_name ~ " " ~ korean_state_char %}
{% endif %}
```

File: web/modules/custom/vote_utility/translations/ko.po [hunk 3]
```diff
@@ -108,34 +111,34 @@ msgid "Find out ways to register"
...
 msgid "Register in person at your local election office. @link on @state_name’s election website."
-msgstr "귀하가 거주하는 지역의 선거 사무소를 직접 방문하여 등록하시기 바랍니다. @state_name 주 선거국 웹사이트에서 @link를  이용하시기 바랍니다."
+msgstr "귀하가 거주하는 지역의 선거 사무소를 직접 방문하여 등록하시기 바랍니다. @state_name 선거국 웹사이트에서 @link를  이용하시기 바랍니다."
 
 msgid "Learn more about voting"
 msgstr "투표에 대해 자세히 알아보기 (영어로 제공)"
 
 msgid "Voter registration is not required in @state_name. @link on @state_name’s election website."
-msgstr "버지니아 주에서는 유권자 등록이 필요하지 않습니다.  @state_name 주 선거국 웹사이트에서 @link를 이용하시기 바랍니다."
+msgstr "버지니아 주에서는 유권자 등록이 필요하지 않습니다.  @state_name 선거국 웹사이트에서 @link를 이용하시기 바랍니다."
```
Comment (rayestrada) [inline web/modules/custom/vote_utility/translations/ko.po:126]: Can you run this through google translate or some other tool? I did and it said "Virginia" it it! We may need a bug ticket to fix this.

Comment (rayestrada) [reply web/modules/custom/vote_utility/translations/ko.po:126]: Also this may say "...in the state" `주에서는`

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: web/modules/custom/vote_utility/translations/ko.po (hunk 3 above)
Comment (SamiraMSadat) [reply web/modules/custom/vote_utility/translations/ko.po:126]: I see the same thing, I will make a bug ticket for this.

File: web/themes/custom/vote_gov/templates/content/page-title.html.twig [hunk 4]
```diff
@@ -15,15 +15,11 @@
...
-    {# set title to alt text for state pages in korean #}
-    {% set title = 'Register to vote in @state_name state' | t ({"@state_name": title}) %}
-  {% else %}
-    {% set title = 'Register to vote in @state_name' | t ({"@state_name": title}) %}
-  {% endif %}
+{% if node_type == "state_territory" and language == 'ko' and currentnode.field_is_state.value %}
+  {% set korean_state_char = "state" | t %}
+  {% set title = title ~ " " ~ korean_state_char %}
 {% endif %}
+{% set title = 'Register to vote in @state_name' | t ({"@state_name": title}) %}
```
Comment (rayestrada) [inline web/themes/custom/vote_gov/templates/content/page-title.html.twig:22]: This solution makes all pages prefaced with "Register to vote in".
Let's rewrite our condition to make sure it's only applied on state/territory pages
```
{% if node_type == "state_territory" %}
  {% if language == 'ko' and currentnode.field_is_state.value %}
    {% set korean_state_char = "state" | t %}
    {% set title = title ~ " " ~ korean_state_char %}
  {% endif %}

  {% set title = 'Register to vote in @state_name' | t ({"@state_name": title}) %}
{% endif %}
```

**Review by rayestrada — CHANGES_REQUESTED** [review]

**Review by rayestrada — APPROVED** [review]

**Pull request merged**